### PR TITLE
fix the problem that memory-swap=-1 is not working for docker command

### PR DESCRIPTION
**Pull request by hqhq**
Signed-off-by: Qiang Huang h.huangqiang@huawei.com

### CODE REVIEW

File: integration-cli/docker_cli_run_test.go [hunk 1]
```diff
@@ -61,6 +61,26 @@ func TestRunEchoStdoutWithMemoryLimit(t *testing.T) {
 }
 
 // "test" should be printed
+func TestRunEchoStdoutWithoutMemoryswapLimit(t *testing.T) {
+	runCmd := exec.Command(dockerBinary, "run", "-m", "16m", "--memory-swap", "-1", "busybox", "echo", "test")
```
Comment (tiborvass) [inline integration-cli/docker_cli_run_test.go]: @hqhq I doubt you need to test for `echo test`, simply replace that with `true`

File: integration-cli/docker_cli_run_test.go [hunk 2]
```diff
@@ -61,6 +61,26 @@ func TestRunEchoStdoutWithMemoryLimit(t *testing.T) {
...
+	out, _, _, err := runCommandWithStdoutStderr(runCmd)
```
Comment (tiborvass) [inline integration-cli/docker_cli_run_test.go]: @hqhq please use `runCommandWithOutput`

File: integration-cli/docker_cli_run_test.go [hunk 3]
```diff
@@ -61,6 +61,26 @@ func TestRunEchoStdoutWithMemoryLimit(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("failed to run container: %v, output: %q", err, out)
+	}
+
+	out = strings.Trim(out, "\r\n")
```
Comment (tiborvass) [inline integration-cli/docker_cli_run_test.go]: @hqhq you can delete lines 71 to 77

File: integration-cli/docker_cli_run_test.go [hunk 4]
```diff
@@ -61,6 +61,26 @@ func TestRunEchoStdoutWithMemoryLimit(t *testing.T) {
...
+
+	if expected := "test"; out != expected {
+		t.Errorf("container should've printed %q but printed %q", expected, out)
+
+	}
+
+	deleteAllContainers()
+
+	logDone("run - echo without memory swap limit")
```
Comment (tiborvass) [inline integration-cli/docker_cli_run_test.go]: @hqhq `run - without memory swap limit`

**Comment by hqhq** [comment]
@tiborvass Updated.
Just curious is the echo not necessary for call cases or just my case? Because I just copy the way from context, not sure why my case would be special?

File: integration-cli/docker_cli_run_test.go [hunk 5]
```diff
@@ -60,6 +60,19 @@ func TestRunEchoStdoutWithMemoryLimit(t *testing.T) {
 	logDone("run - echo with memory limit")
 }
 
+// should run without memory swap
+func TestRunWithoutMemoryswapLimit(t *testing.T) {
+	runCmd := exec.Command(dockerBinary, "run", "-m", "16m", "--memory-swap", "-1", "busybox", "true")
+	out, _, err := runCommandWithOutput(runCmd)
+	if err != nil {
+		t.Fatalf("failed to run container: %v, output: %q", err, out)
```
Comment (tiborvass) [inline integration-cli/docker_cli_run_test.go]: no need to print `err` since it does not hold useful information.

Comment (hqhq) [inline integration-cli/docker_cli_run_test.go]: Done.

**Comment by tiborvass** [comment]
@hqhq Thanks! Sorry I didn't see you took the `echo test` from the other tests. I'm not sure why they were there in the first place though, I still think `true` makes more sense. @unclejack can you confirm?

**Comment by jfrazelle** [comment]
can confirm the test fails on master LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**